From ade49c4b574f33b6703de883ca531fd4b7058511 Mon Sep 17 00:00:00 2001
From: Andrei Danaila <andrei.danaila@intel.com>
Date: Mon, 18 Apr 2016 10:04:05 -0700
Subject: [PATCH 2/2] platform: Add ap21x1 devices to platform

Add two AP21x1 power switches as platform devices
on the Haven Park platform.

Signed-off-by: Andrei Danaila <andrei.danaila@intel.com>
---
 arch/x86/platform/intel-ce2700/Makefile     |    1 +
 arch/x86/platform/intel-ce2700/haven_park.c |   61 +++++++++++++++++++++++++++
 2 files changed, 62 insertions(+)
 create mode 100644 arch/x86/platform/intel-ce2700/haven_park.c

--- a/arch/x86/platform/intel-ce2700/Makefile
+++ b/arch/x86/platform/intel-ce2700/Makefile
@@ -1 +1,2 @@
 obj-$(CONFIG_X86_INTEL_CE2700) += intel-ce2700.o
+obj-$(CONFIG_X86_INTEL_HAVEN_PARK) += haven_park.o
--- /dev/null
+++ b/arch/x86/platform/intel-ce2700/haven_park.c
@@ -0,0 +1,61 @@
+/*
+ * haven-park.c Board file for Haven Park platform
+ * Copyright (c) 2016 Intel Corporation. All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ */
+
+#include <linux/power/ap21x1_switch.h>
+#include <linux/platform_device.h>
+#include <linux/module.h>
+
+static struct ap21x1_data ap21x1[] = {
+	{
+		/* Controlling - USB Type A connector */
+		.gpio_vin = 172, /* EXT_INT4_ARM11_RTCK_3P3 */
+		.gpio_vin_assert_value = 0,
+		.gpio_oc = 280,  /* USB_OC1_B_1P8 */
+		.gpio_oc_assert_value = 0,
+		.gpio_oc_irq = 0,
+	},
+	{
+		/* Controlling - USB Type C connector */
+		.gpio_vin = 237 /* SC0_CLK_3P3 */,
+		.gpio_vin_assert_value = 0,
+		.gpio_oc = 281, /* USB_OC0_B_1P8 */
+		.gpio_oc_assert_value = 0,
+		.gpio_oc_irq = 0,
+	},
+};
+
+static struct platform_device ap21x1_devices[] = {
+	{
+		.name	= "ap21x1",
+		.id	= -1,
+		.dev	= {
+			.platform_data	= &ap21x1[0],
+		},
+	},
+	{
+		.name	= "pxa21x1",
+		.id	= -1,
+		.dev	= {
+			.platform_data	= &ap21x1[1],
+		},
+	},
+};
+
+static int __init hp_platform_init(void)
+{
+	return platform_add_devices(ap21x1_devices,
+			ARRAY_SIZE(ap21x1_devices));
+}
+
+device_initcall(hp_platform_init);
